autoware_cmake/cmake/autoware_package.cmake

```diff
@@ -0,0 +1,45 @@
+# Copyright 2022 The Autoware Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+macro(autoware_package)
```

Comment:
This macro can be changed to a function, it'd be less invasive

Reply:
@esteve Thank you for your comment! :pray: 
Okay, I'll consider that. Also, to check the behavior, I'll create a pull request in `autoware.universe`.

Reply:
I learned the differences between `macro` and `function`.
Considering the current contents, I guess `macro` is appropriate, isn't it?
If we use `function`, for example, `CMAKE_CXX_STANDARD` will be reset after the function call.

If we write without `macro` nor `function`, it is executed after `find_package(autoware_cmake REQUIRED)`, but I feel it's a bit confusing.
https://github.com/autowarefoundation/autoware.universe/blob/b0dcc90d6dcbbb465fe615e07ec1552858cd818c/common/autoware_auto_cmake/cmake/autoware_auto_cmake.cmake#L19

@esteve What do you think about it?

Reply:
@kenji-miyake you're right, we need this to be a `macro` so that it modifies the surrounding code (i.e. `CMAKE_CXX_STANDARD`), thanks for looking into it.